lerobot/scripts/eval.py

```diff
@@ -556,10 +554,12 @@ def main(
     assert isinstance(policy, nn.Module)
     policy.eval()
 
-    with torch.no_grad(), torch.autocast(device_type=device.type) if hydra_cfg.use_amp else nullcontext():
+    if accelerator:
+        policy = accelerator.prepare_model(policy)
+    with torch.no_grad():
         info = eval_policy(
             env,
-            policy,
+            policy if accelerator is None else accelerator.unwrap_model(policy),
```

Comment:
Wait, but if you pass the unwrapped policy, are you still using the accelerator when you call the policy in rollout?

Reply:
`.prepare` handles the autocast, it adds a a context to the forward of the model. I added the `keep_fp32_wrapper` flag

Reply:
You did? I'm not seeing it here. Isn't it meant to be a kwarg of `unwrap_model`?

Reply:
actually no, sorry I was wrong. Removing the `fp32_wrapper` will only cause the outputs of the farward to stay in the `fp16` type, which is not the behaviour we'd want so we should keep it the default `fp32_wrapper=True.` unwrapping the model will only take it out of its distributed wrappers.
The rest of the context with `autocast` is handled when calling `.prepare`

Reply:
Update after discussing it with the Accelerate team lead : `keep_fp32_wrapper` does allow to keep the autocast context, so I added that.